### ISSUE TITLE
redis pods are restarting again and again 

### ISSUE THREAD

**Issue by abhishekgupta2205** [bitnami/charts#14394]
### Name and Version

bitnami/redis , latest

### What steps will reproduce the bug?

helm install redis
sentinel mode is enabled
1 master + 2 slave architecture

### What is the expected behavior?

redis pods should not restart again and again .

### What do you see instead?

pods are restarting .

### Additional information

logs of pods are 
![Screenshot from 2023-01-17 11-46-37](https://user-images.githubusercontent.com/68244292/212823813-c7330572-69f5-4b87-879b-6b75e800977a.png)
![Screenshot from 2023-01-17 11-46-37](https://user-images.githubusercontent.com/68244292/212823831-c09c801d-ed67-44e3-a218-ac5f61bb5e6e.png)

**Comment by abhishekgupta2205** [comment]
![image](https://user-images.githubusercontent.com/68244292/212824049-b394b493-fad7-4eae-bbc7-e67207abf62b.png)

**Comment by abhishekgupta2205** [comment]
![image](https://user-images.githubusercontent.com/68244292/212824655-e1694c4d-154f-48ce-8914-ac74d6e3e7ae.png)

**Comment by abhishekgupta2205** [comment]
![image](https://user-images.githubusercontent.com/68244292/212899994-dd01fbaa-a985-4c7b-adb7-98b16d37c716.png)
i think this is causing restart . but i dont know why it is happening

**Comment by abhishekgupta2205** [comment]
@CeliaGMqrz anything on this

**Comment by PovilasKM** [comment]
We are experiencing the same issue. We've noticed that our exit code is 137 which means that our pod or container is out of memory but dashboards suggest that our memory usage never goes above 60%. Our sentinel logs are the same as yours. There was a similar ticket created a year ago with quite a lot of troubleshooting but it never got anywhere.

**Comment by abhishekgupta2205** [comment]
@marcosbc @jonathon2nd did you guys find any solution of it?

**Comment by CeliaGMqrz** [comment]
Hi @abhishekgupta2205, sorry for the delay.

Thanks for your report. I can't reproduce the issue, but I think the resources you provide may be involved. Are you monitoring your cluster? Have the pods enough resources? You can specify the resources in `values.yaml`

```yaml
  ## Redis&reg; replicas resource requests and limits
  ## ref: https://kubernetes.io/docs/user-guide/compute-resources/
  ## @param replica.resources.limits The resources limits for the Redis&reg; replicas containers
  ## @param replica.resources.requests The requested resources for the Redis&reg; replicas containers
  ##
  resources:
    # We usually recommend not to specify default resources and to leave this as a conscious
    # choice for the user. This also increases chances charts run on environments with little
    # resources, such as Minikube. If you do want to specify resources, uncomment the following
    # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
    limits: {}
    #   cpu: 250m
    #   memory: 256Mi
    requests: {}
    #   cpu: 250m
    #   memory: 256Mi
```

**Comment by abhishekgupta2205** [comment]
i dont think resources given to pod is issue here as i have tried it by increasing it to around 1gb memory for a single pod and still it is showing same errors.

**Comment by CeliaGMqrz** [comment]
Hi @abhishekgupta2205, 

Thanks for your comments, however, we would require more details on how you deployed the chart because we can not reproduce it. Which values did you set? Which is your Kubernetes platform? What are your steps to reproduce it? Do you have a script to load the data?  Does this problem appear immediately or after some time has passed?

**Comment by abhishekgupta2205** [comment]
i think this is same issue as 10547 which remained unsolved till now
which was opened by @jonathon2nd

**Comment by abhishekgupta2205** [comment]
i have tried it on minikube as well as on-premises kubernetes cluster having 3 nodes and 3 pods where each pod is scheduled on a differnt node. it works fine for some time and then tilting starts

**Comment by CeliaGMqrz** [comment]
Hi @abhishekgupta2205, 

Thanks for the detailed information, our team has provided a possible solution.
Can you try the changes proposed in this PR [#14599](https://github.com/bitnami/charts/pull/14599)? 

We appreciate your feedback.

**Comment by abhishekgupta2205** [comment]
@CeliaGMqrz can you exactly tell what i need to change to solve this issue

**Comment by fmulero** [comment]
Hi @abhishekgupta2205 

Please don't duplicate issues, I've just closed #14529 because it is just a copy of this one.

This issue also is pretty similar to #9689, have you take it glance? There are several discussions about the configuration and possible solutions, also the PR #14599 mentioned by @CeliaGMqrz tries to solve this issue allowing the use of IPs instead of hostnames when configuring the different nodes in the deployment.

**Comment by abhishekgupta2205** [comment]
yes my issue is same as 9689 but there also no solution was provided , so thats why i opened a new issue .

**Comment by abhishekgupta2205** [comment]
i am using k8s version 1.24 ,cni version- 0.3.1, plugin type calico

**Comment by fmulero** [comment]
Closed as duplicate of #9689
